lib/r10k/environment/base.rb

```diff
@@ -124,28 +151,37 @@ def accept(visitor)
   end
 
   def deploy
-    puppetfile.load(@overrides.dig(:environments, :default_branch_override))
+    loaded_content = @loader.load
```

Comment:
It's weird to have `modules` call `@loader.load` as well as `deploy`. Can we factor that into a shared method?